### PR TITLE
Feature/testing clinics view

### LINKED ISSUE #37: Cypress testing for Clinics page

### PR DESCRIPTION
**Pull request by Rallen13**
** resolves #37  **

** Describe the functionality to be tested: **

- full test suite covering the clinic's view, search, and filtering
- cypress commands setup and stubbing
- some styling updates
- no result message for empty sections when searching

** Giphy of how you feel submitting this PR: **

### CODE REVIEW

File: cypress/e2e/clinicsView.cy.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import clinics from '../fixtures/clinics.json';
+
+describe('Clinics View', () => {
+  beforeEach(() => {
+    cy.interceptGQL(
+      'https://roed-to-aide-be.herokuapp.com/graphql?api_key=ca912ed1df0d1c0f014ec94e3c731881',
+      'GetClinics',
```
Comment (lkarpins) [inline cypress/e2e/clinicsView.cy.js:7]: THAT WAS 🤯🤯🤯

File: src/Components/ClinicCardContainer/ClinicCardContainer.js [hunk 2]
```diff
@@ -45,43 +70,78 @@ const ClinicCardContainer = ({ states, toggleSelected, filteredResults }) => {
...
-            {authorizedClinics}
-          </div>
+          {authorized.length ? (
+            <div
+              className="card-container authorized-card-container"
+              data-cy="authorized-card-container"
+            >
+              {authorizedClinics}
+            </div>
+          ) : (
```
Comment (lkarpins) [inline src/Components/ClinicCardContainer/ClinicCardContainer.js:102]: Genius! 🧠

File: src/Views/ClinicsView/ClinicsView.js [hunk 3]
```diff
@@ -75,25 +76,43 @@ const ClinicsView = () => {
...
   return (
-    <>
-      <h1>Find a safe clinic</h1>
+    <main className="clinics-view-container">
+      <h1 data-cy="clinics-view-heading">Find a safe clinic</h1>
       <div className="heading-container">
         <ClinicKeyBlock />
         <div>
+          <p className="clinics-view-disclaimer" data-cy="clinics-view-disclaimer">
+            As this is an MVP project, we currently only have results for
```
Comment (lkarpins) [inline src/Views/ClinicsView/ClinicsView.js:90]: Great addition!

**Review by lkarpins — APPROVED** [review]
That was  A LOOOOOOOT! Awesome work, Rachel!! ⭐️

File: cypress/e2e/clinicsView.cy.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
+import clinics from '../fixtures/clinics.json';
+
+describe('Clinics View', () => {
+  beforeEach(() => {
+    cy.interceptGQL(
+      'https://roed-to-aide-be.herokuapp.com/graphql?api_key=ca912ed1df0d1c0f014ec94e3c731881',
```
Comment (catlynbowles) [inline cypress/e2e/clinicsView.cy.js:6]: I AM SO GLAD YOU FIGURED THIS OUT. GET THAT BONE!! 🦴

File: cypress/e2e/clinicsView.cy.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+      cy.dataCy('misinformation-card-container').within(() => {
+        cy.dataCy('clinic-card').within(() => {
+          cy.dataCy('clinic-name').should('exist');
+          cy.dataCy('clinic-address').should('exist');
+          cy.dataCy('clinic-city-state-zip').should('exist');
+        });
+      });
+    });
+  });
+});
```
Comment (catlynbowles) [inline cypress/e2e/clinicsView.cy.js:185]: whoahhhh so much testing!!! 🙌🏼

**Review by catlynbowles — APPROVED** [review]

File: cypress/e2e/clinicsView.cy.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+  afterEach(() => {
+    cy.visit('http://localhost:3000/clinics');
+  });
+
+  it('should contain a heading', () => {
+    cy.dataCy('clinics-view-heading').contains('Find a safe clinic');
+  });
+
+  it('should contain an authorized key block', () => {
+    cy.dataCy('clinic-key-block-authorized').within(() => {
```
Comment (irmakerdem) [inline cypress/e2e/clinicsView.cy.js:23]: So much nesting. So much power 🔋

File: src/Components/ClinicCardContainer/ClinicCardContainer.js [hunk 7]
```diff
@@ -1,19 +1,43 @@
...
+  toggleSelected,
+  filteredResults,
+  searchInput,
+}) => {
+  const [authorized, setAuthorized] = useState([]);
+  const [misinformation, setMisinformation] = useState([]);
+  const [resources, setResources] = useState([]);
 
   useEffect(() => {
     if (filteredResults.length > 0) {
```
Comment (irmakerdem) [inline src/Components/ClinicCardContainer/ClinicCardContainer.js:17]: could all of the `state.authorizedClinics.length > 0` in this hook become `state.authorizedClinics.length` and still work?

**Review by irmakerdem — APPROVED** [review]

**Pull request merged**